t/ingredients.t

```diff
@@ -8,2349 +8,277 @@ use utf8;
-is_deeply ($product_ref->{labels_tags}, undef) or diag explain $product_ref->{labels_tags};
-
-is_deeply(
-	$product_ref->{ingredients},
+	# a label and multiple origins in parenthesis -- does not work yet
 	[
-		{	'id'         => 'en:green-tea',
-			'labels'     => 'en:organic',
-			'rank'       => 1,
-			'text'       => "vihre\x{e4} tee",
-			'vegan'      => 'yes',
-			'vegetarian' => 'yes'
-		},
-		{	'id'         => 'en:milk',
-			'labels'     => 'en:organic',
-			'rank'       => 2,
-			'text'       => 'maito',
-			'vegan'      => 'no',
-			'vegetarian' => 'yes'
-		},
-		{	'id'         => 'en:malted-barley',
-			'labels'     => 'en:organic',
-			'rank'       => 3,
-			'text'       => 'ohramallas',
-			'vegan'      => 'yes',
-			'vegetarian' => 'yes'
+		"fr-label-and-multiple-origins",
+		{
+			lc => "fr",
+			ingredients_text => "oeufs (d'élevage au sol, Suisse, France)",
 		}
 	],
-) or diag explain $product_ref;
-
-
-$product_ref = {
-	lc => "fr",
-	ingredients_text => "oeufs (d'élevage au sol, Suisse, France)",
-};
-
-extract_ingredients_from_text($product_ref);
-
-delete_ingredients_percent_values($product_ref->{ingredients});
-delete $product_ref->{ingredients_percent_analysis};
-
-
-is ($product_ref->{labels}, undef) or diag explain $product_ref->{labels};
-is_deeply ($product_ref->{labels_tags}, undef) or diag explain $product_ref->{labels_tags};
-
-is_deeply (
-	$product_ref->{ingredients},
 
+	# Do not mistake single letters for labels, bug #3300
 	[
+		"xx-single-letters",
 		{
-		'has_sub_ingredients' => 'yes',
-		'id' => 'en:egg',
-		'ingredients' => [
-			{
-				'id' => "fr:d'\x{e9}levage au sol",
-				'text' => "d'\x{e9}levage au sol"
-			},
-			{
-				'id' => 'fr:Suisse',
-				'text' => 'Suisse'
-			},
-			{
-				'id' => 'fr:France',
-				'text' => 'France'
-			}
-		],
-			'rank' => 1,
-			'text' => 'oeufs',
-			'vegan' => 'no',
-			'vegetarian' => 'yes'
-		},
-		{
-			'id' => "fr:d'\x{e9}levage au sol",
-			'text' => "d'\x{e9}levage au sol"
-		},
-		{
-			'id' => 'fr:Suisse',
-			'text' => 'Suisse'
-		},
-		{
-			'id' => 'fr:France',
-			'text' => 'France'
+			lc => "fr",
+			ingredients_text => "a,b,c,d,e,f,g,h,i,j,k,l,m,n,o,p,q,r,s,t,u,v,w,x,y,z,0,1,2,3,4,5,6,7,8,9,10,100,1000,vt,leaf,something(bio),somethingelse(u)",
 		}
 	],
-
-) or diag explain $product_ref;
-
-
-# Do not mistake single letters for labels, bug #3300
-
-$product_ref = {
-	lc => "fr",
-	ingredients_text => "a,b,c,d,e,f,g,h,i,j,k,l,m,n,o,p,q,r,s,t,u,v,w,x,y,z,0,1,2,3,4,5,6,7,8,9,10,100,1000,vt,leaf,something(bio),somethingelse(u)",
-};
-
-extract_ingredients_from_text($product_ref);
-
-delete_ingredients_percent_values($product_ref->{ingredients});
-delete $product_ref->{ingredients_percent_analysis};
-
-
-is ($product_ref->{labels}, undef) or diag explain $product_ref->{labels};
-is_deeply ($product_ref->{labels_tags}, undef) or diag explain $product_ref->{labels_tags};
+);
+
+my $testdir = "ingredients";
+
+if (! -e "test_results") {
```

Comment:
Maybe place this in a well-known (`/tmp`?) or configured directory (`$data-root`). Otherwise, this depends on the exact path where the test is run from. Also, this dir should probably be `.gitignore`.